src/query-string/parse.js

```diff
@@ -0,0 +1,48 @@
+function addToObj(parameters, key, value) {
+  const innerObj = parameters;
+
+  if (innerObj[key] === undefined) {
+    if (value.length === 0) {
+      innerObj[key] = null;
+    } else {
+      innerObj[key] = value;
+    }
+  } else if (typeof innerObj[key] === 'string') {
+    innerObj[key] = [innerObj[key], value];
+  } else {
+    innerObj[key].push(value);
+  }
+
+  return innerObj;
+}
+
+/**
+ * Parse a string url and return the object with
+ * the parameters and values
+ *
+ * @memberof module:@linx/commons-js/query-string
+ * @method parse
+ * @param {string} url The url to be parsed
+ * @returns {object} parameters The object to be returned with the parameters
+ */
+export function parse(url) {
+  const decodedUrl = decodeURIComponent(url);
+  const query = decodedUrl.split('?')[1];
+
+  if (!query) {
+    return null;
+  }
```

Comment:
Três coisas nesse bloco aqui:

1. Não precisa fazer o split pelo `?`. Vamos fazer a função funcionar apenas para o queryString de fato. Esse é o padrão usado no node, então quem for usar vai ter que fazer o split e passar só o querystring como parâmetro;

2. Checa se o url é uma string, se não for, retorna um objeto vazio, é melhor do que null, pois mantemos a consistência da resposta.

```javascript
if (typeof url !== 'string') {
  return {};
}
```